packaging/windows/install_script.nsis

```diff
@@ -0,0 +1,278 @@
+# adds uninstall information to the registry for Add/Remove Programs
+
+!define APPNAME "Grafana Agent"
+!define DESCRIPTION "The Grafana Agent collects observability data and sends it to a compatible remote write endpoint"
+# These will be displayed by the "Click here for support information" link in "Add/Remove Programs"
+!define HELPURL "https://github.com/grafana/agent/discussions" # "Support Information" link
+!define UPDATEURL "https://github.com/grafana/agent/releases" # "Product Updates" link
+!define ABOUTURL "https://github.com/grafana/agent" # "Publisher" link
+!include LogicLib.nsh
+
+RequestExecutionLevel admin #Require admin rights on NT6+ (When UAC is turned on)
+
+InstallDir "$PROGRAMFILES\${APPNAME}"
+
+# This will be in the installer/uninstaller's title bar
+Name "${APPNAME} ${VERSION}"
+Icon "logo.ico"
+outFile "/app/grafana-agent-installer.exe"
+
+!include LogicLib.nsh
+!include nsDialogs.nsh
+!include FileFunc.nsh
+
+
+# Everything must be global Vars
+
+# These are handles to dialogs
+Var Dialog
+Var RemoteWriteLabel
+Var RemoteWriteText
+
+Var RemoteWriteUserLabel
+Var RemoteWriteUserText
+
+Var RemoteWritePasswordLabel
+Var RemoteWritePasswordText
+
+Var EnableExporterLabel
+Var EnableExporterCheck
+
+# These are strings that hold the values
+
+Var RemoteUrl
+Var RemoteUser
+Var RemotePassword
+Var EnableExporterValue
+
+# Silent install params
+
+Var PassedInParameters
+
+Page directory
+Page custom remoteWriteInfo remoteWriteInfoLeave
+Page custom enableWindowsExporter enableWindowsExporterLeave
+Page instfiles
+
+
+
+!macro VerifyUserIsAdmin
+UserInfo::GetAccountType
+pop $0
+${If} $0 != "admin" #Require admin rights on NT4+
+        messageBox mb_iconstop "Administrator rights required!"
+        setErrorLevel 740 #ERROR_ELEVATION_REQUIRED
+        quit
+${EndIf}
+!macroend
+
+
+Section SilentInstaller
+    IfSilent ThisIsSilent RunInstaller
+    ThisIsSilent:
+        ${GetParameters} $PassedInParameters
+        ${GetOptions} $PassedInParameters "/RemoteURL" $RemoteUrl
+        ${GetOptions} $PassedInParameters "/RemoteUser" $RemoteUser
+        ${GetOptions} $PassedInParameters "/RemotePassword" $RemotePassword
+        ${GetOptions} $PassedInParameters "/EnableExporter" $EnableExporterValue
+        Call Install
+        # TODO add error conditions
+        Return
+    RunInstaller:
+        Return
+SectionEnd
+
+
+Section "install"
+    IfSilent ThisIsSilent RunInstaller
+    ThisIsSilent:
+        Return
+    RunInstaller:
+        Call Install
+SectionEnd
+
+Function remoteWriteInfo
+    nsDialogs::Create 1018
+    Pop $Dialog
+
+    ${If} $Dialog == error
+        Abort
+    ${EndIf}
+    ${NSD_CreateLabel} 0 0 100% 12u "Enter URL for remote write"
+    Pop $RemoteWriteLabel
+
+    ${NSD_CreateText} 0 13u 100% 12u ""
+    Pop $RemoteWriteText
+
+    ${NSD_CreateLabel} 0 26u 100% 12u "Enter User for remote write"
+    Pop $RemoteWriteUserLabel
```

Comment:
Do we need to pop the handle off the stack if we're not going to use it?

Reply:
Moot point now, but yes. Though to be fair I could have used one of the registers $0 for clarity, thats its throwaway.